assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java

```diff
@@ -264,7 +264,15 @@ private void configureSingleUserMode() {
     bind(org.eclipse.che.security.oauth.OAuthAuthenticationService.class);
   }
 
-  private void configureMultiUserMode() {
+  private void configureMultiUserMode(
+      Map<String, String> persistenceProperties, String infrastructure) {
+    if (OpenShiftInfrastructure.NAME.equals(infrastructure)
```

Comment:
This hardcoding of infrastructure names makes Che less extensibility friendly. I would recommend using another approach where Che deployment can specify environment variable to configure JGROUPS. @mshaposhnik @skabashnyuk @sleshchenko WDYT?

Reply:
There is a lot of ohter places where it is configured in a such way.  It is not easy to refactor them all. Probably it's another big issue.

Reply:
@garagatyi how do you propose to implement in another way?

Reply:
@skabashnyuk having an environment variable that enables what now is enabled by the name of the active infrastructure.